applications/CompressiblePotentialFlowApplication/custom_utilities/potential_flow_utilities.cpp

```diff
@@ -41,9 +41,148 @@ array_1d<double, NumNodes> GetPotentialOnNormalElement(const Element& rElement)
     return potentials;
 }
 
+template <int Dim, int NumNodes>
+BoundedVector<double, 2 * NumNodes> GetPotentialOnWakeElement(
+    const Element& rElement, const array_1d<double, NumNodes>& rDistances)
+{
+    array_1d<double, NumNodes> upper_potentials;
+    upper_potentials = GetPotentialOnUpperWakeElement<Dim, NumNodes>(rElement, rDistances);
+
+    array_1d<double, NumNodes> lower_potentials;
+    lower_potentials = GetPotentialOnLowerWakeElement<Dim, NumNodes>(rElement, rDistances);
+
+    BoundedVector<double, 2 * NumNodes> split_element_values;
+    for (unsigned int i = 0; i < NumNodes; i++) {
+        split_element_values[i] = upper_potentials[i];
+        split_element_values[NumNodes + i] = lower_potentials[i];
+    }
+
+    return split_element_values;
+}
+
+template <int Dim, int NumNodes>
+BoundedVector<double, NumNodes> GetPotentialOnUpperWakeElement(
+    const Element& rElement, const array_1d<double, NumNodes>& rDistances)
+{
+    array_1d<double, NumNodes> upper_potentials;
+    const auto r_geometry = rElement.GetGeometry();
+
+    for (unsigned int i = 0; i < NumNodes; i++){
+        if (rDistances[i] > 0){
+            upper_potentials[i] = r_geometry[i].FastGetSolutionStepValue(VELOCITY_POTENTIAL);
+        }
+        else{
+            upper_potentials[i] = r_geometry[i].FastGetSolutionStepValue(AUXILIARY_VELOCITY_POTENTIAL);
+        }
+    }
+
+    return upper_potentials;
+}
+
+template <int Dim, int NumNodes>
+BoundedVector<double, NumNodes> GetPotentialOnLowerWakeElement(
+    const Element& rElement, const array_1d<double, NumNodes>& rDistances)
+{
+    array_1d<double, NumNodes> lower_potentials;
+    const auto r_geometry = rElement.GetGeometry();
+
+    for (unsigned int i = 0; i < NumNodes; i++){
+        if (rDistances[i] < 0){
+            lower_potentials[i] = r_geometry[i].FastGetSolutionStepValue(VELOCITY_POTENTIAL);
+        }
+        else{
+            lower_potentials[i] = r_geometry[i].FastGetSolutionStepValue(AUXILIARY_VELOCITY_POTENTIAL);
+        }
+    }
+
+    return lower_potentials;
+}
+
+template <int Dim, int NumNodes>
+array_1d<double, Dim> ComputeVelocity(const Element& rElement)
+{
+    const int wake = rElement.GetValue(WAKE);
+
+    if (wake == 0)
```

Comment:
```suggestion
    if (wake == rElement.GetValue(WAKE))
```

Reply:
I think I don't get this, did you mean:

`if (rElement.GetValue(WAKE) == 0)`